Automation component

marinmitev opened the pull request:
Automation API and JSON files - initial contribution by ProSyst

Author: Marin Mitev <marin.mitev@gmail.com>
Also-by: Jordan Mihailov <j.mihailov@prosyst.com>, Vasil Ilchev <v.ilchev@prosyst.bg>, Ana Dimova <a.dimova@prosyst.bg>

kaikreuzer commented:
Marin, thanks a lot for this PR! We will make look at it asap and give you feedback!

kaikreuzer commented:
Thanks for squashing, Marin!
Unfortunately, the CLA validation fails - did you use by mistake a wrong e-mail address for signing?
Please also make sure to list any colleagues who were involved as "Also-by", see https://wiki.eclipse.org/Development_Resources/Contributing_via_Git#The_Commit_Record.

Next step is for me then to create a CQ, i.e. to pass the code on to the IP team at Eclipse. Thanks again!

marinmitev commented:
Hi Kai, it seems I have not completed the CLA process, and I have just did it
Please check again
Thanks,
Marin

kaikreuzer commented:
Thanks Marin,

but unfortunately, there are still two issues:
1. There are now 3 commits instead of 1 - you will have to squash it again
2. All "Also-by" contributors need to have a signed CLA - this does not seem to be the case.

Cheers,
Kai

dnobel commented:
Moreover the "Signed-off-by <author>" line is missing. The format must be:
```
Some commit message

Also-by: Some Bodyelse1 <somebodyelse1@nowhere.com>
Also-by: Some Bodyelse2 <somebodyelse2@nowhere.com>
Signed-off-by: Some Body <somebody@somewhere.com>
```

marinmitev commented:
Hi again,
squashed all commits to one;
splitted the co-authors in separate lines;
added signed off account;
and created eclipse accounts with signed CLA agreement
Now I hope everything is OK,
Best regards,
Marin

dnobel commented:
Thanks Marin! As the message "All is well — All committers passed Eclipse CLA and Signed-off-by validation." indicates, now everything is fine (I also rechecked the CLA for every author manually). So we can merge the PR.

kaikreuzer commented:
wonderful, thanks!
But @dnobel, we cannot merge yet, I now first have to create a CQ for this PR!

dnobel commented:
Sorry, you are right. Definitely more than 1000 lines

kaikreuzer commented on bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/template/TemplateProvider.java:
```diff
@@ -0,0 +1,45 @@
+ * http://www.prosyst.com
+ * All rights reserved.
+ *
+ * This software is the confidential and proprietary information
```
Could you fix the license header to be EPL?

kaikreuzer commented on bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/type/ModuleTypeProvider.java:
```diff
@@ -0,0 +1,46 @@
+ * http://www.prosyst.com
+ * All rights reserved.
+ *
+ * This software is the confidential and proprietary information
```
Could you fix the license header to be EPL?

kaikreuzer commented on bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/Connection.java:
```diff
@@ -0,0 +1,91 @@
+ * http://www.prosyst.com
+ * All rights reserved.
+ *
+ * This software is the confidential and proprietary information
```
Could you fix the license header to be EPL?

kaikreuzer commented:
@marinmitev , when creating the CQ I stumbled across three license headers that are not correct - so I have to ask you for yet another fix and a squash!

marinmitev commented:
Kai, sorry for that - we have overlooked these 3 files, now it should be ok

kaikreuzer commented:
Thanks again! I have now created CQ9374, hence we will now have to wait for the approval of the IP team.

kaikreuzer commented:
Good news everybody! We have just received the approval to merge this PR in accordance with [Parallel
IP](https://wiki.eclipse.org/Development_Resources/HOWTO/Parallel_IP_Process).
